File: backend/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema(
  {
    productId: {
      type: String,
      unique: true,
      required: true,
    },
    name: {
      type: String,
      required: [true, 'Product name is required'],
      trim: true,
    },
    category: {
      type: String,
      enum: ['MILK', 'MILK_PRODUCT', 'VEGETABLE', 'CHICKEN', 'MEAT', 'OTHER'],
      required: true,
    },
    description: { type: String, trim: true },
    unit: { type: String, required: true, default: 'L' }, // L, kg, 500g, 250g, bunch
    price: { type: Number, required: true },
    farmer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    location: { type: String, required: true, default: 'Dindigul' },
    emoji: { type: String, default: '🥛' },

    // Specialized Milk & Dairy Attributes
    fatContent: { type: String, default: null }, // e.g. "4.5% Fat"
    snfContent: { type: String, default: null }, // e.g. "8.5% SNF"
    milkingSlot: { type: String, default: null }, // e.g. "4:30 AM Milking"
    packaging: { type: String, default: 'Eco Glass Bottle' },
    isOrganic: { type: Boolean, default: true },
    isA2: { type: Boolean, default: false },

    status: {
      type: String,
      enum: ['AVAILABLE', 'LIMITED', 'SOLD_OUT', 'INACTIVE'],
      default: 'AVAILABLE',
    },
    rating: { type: Number, default: 4.8 },
    reviewCount: { type: Number, default: 50 },
  },
  { timestamps: true }
);

productSchema.index({ name: 'text', productId: 'text', location: 'text' });

module.exports = mongoose.model('Product', productSchema);

File: backend/Models/Product.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace DailyMarts.Api.Models;

[BsonIgnoreExtraElements]
public class Product
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

    [BsonElement("productId")]
    public string ProductId { get; set; } = string.Empty;

    [BsonElement("name")]
    public string Name { get; set; } = string.Empty;

    [BsonElement("category")]
    public string Category { get; set; } = "MILK"; // MILK, MILK_PRODUCT, VEGETABLE, CHICKEN, MEAT, OTHER

    [BsonElement("description")]
    public string Description { get; set; } = string.Empty;

    [BsonElement("unit")]
    public string Unit { get; set; } = "L";

    [BsonElement("price")]
    public decimal Price { get; set; }

    [BsonElement("farmer")]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Farmer { get; set; } = string.Empty;

    [BsonElement("location")]
    public string Location { get; set; } = "Dindigul";

    [BsonElement("emoji")]
    public string Emoji { get; set; } = "🥛";

    // Milk & Dairy Attributes
    [BsonElement("fatContent")]
    public string? FatContent { get; set; }

    [BsonElement("snfContent")]
    public string? SnfContent { get; set; }

    [BsonElement("milkingSlot")]
    public string? MilkingSlot { get; set; }

    [BsonElement("packaging")]
    public string Packaging { get; set; } = "Eco Glass Bottle";

    [BsonElement("isOrganic")]
    public bool IsOrganic { get; set; } = true;

    [BsonElement("isA2")]
    public bool IsA2 { get; set; } = false;

    [BsonElement("status")]
    public string Status { get; set; } = "AVAILABLE"; // AVAILABLE, LIMITED, SOLD_OUT, INACTIVE

    [BsonElement("rating")]
    public double Rating { get; set; } = 4.8;

    [BsonElement("reviewCount")]
    public int ReviewCount { get; set; } = 50;

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Populated navigation property for API JSON serialization
    [BsonIgnore]
    public User? FarmerDetails { get; set; }
}
